src/alphafold_impact/pipelines/data_collection_nih_funding_opportunities/nodes.py

```diff
@@ -0,0 +1,158 @@
+"""
+This module contains functions for fetching NIH funding opportunities data.
+
+Functions:
+    collect_nih_funding_opportunities: Downloads and saves NIH
+        funding opportunities to S3.
+"""
+
+import logging
+from datetime import datetime
+import time
+from typing import Dict, Union
+import requests
+from requests.adapters import HTTPAdapter, Retry
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+
+def _get_params(end_date: str, perpage: int) -> Dict[str, any]:
+    """
+    Returns the parameters for the NIH grants API requests with
+    specified end date and records per page.
+
+    Args:
+        end_date: The end date for the date range filter in
+            the format 'DDMMYYYY'.
+        perpage: Number of records per page.
+
+    Returns:
+        A dictionary of parameters for the API request.
+    """
+    return {
+        "perpage": perpage,
+        "sort": "reldate:desc",
+        "type": "active,expired,activenosis,expirednosis",
+        "parentic": "all",
+        "primaryic": "all",
+        "activitycodes": "all",
+        "doctype": "all",
+        "parentfoa": "all",
+        "daterange": f"01021991-{end_date}",
+        "clinicaltrials": "all",
+        "fields": "all",
+        "spons": "true",
+        "query": "",
+    }
+
+
+def _make_api_request(url: str, params: Dict[str, any]) -> Union[Dict[str, any], None]:
+    """
+    Creates a session, makes an API request, and returns the
+    JSON response.
+
+    Args:
+        url: The API endpoint URL.
+        params: Parameters for the API request.
+
+    Returns:
+        The JSON response as a dictionary, or None if the
+            request fails.
+    """
+    session = requests.Session()
```

Comment:
We should consider just creating a single, non-pipeline specific function for these requests. So much code redundancy

Reply:
I've added a new issue relating to this